deepchem/models/torch_models/cgcnn.py

```diff
@@ -0,0 +1,255 @@
+
+  This class was implemented using DGLGraph methods.
+  Please confirm how to use DGLGraph methods from below link.
+  See: https://docs.dgl.ai/en/0.4.x/tutorials/models/1_gnn/9_gat.html
+  """
+
+  def __init__(self,
+               hidden_node_dim: int,
+               edge_dim: int,
+               batch_norm: bool = True):
+    """
+    Parameters
+    ----------
+    hidden_node_dim: int
+      The length of the hidden node feature vectors.
+    edge_dim: int
+      The length of the edge feature vectors.
+    batch_norm: bool, default True
+      Whether to apply batch normalization or not.
+    """
+    super(CGCNNLayer, self).__init__()
+    z_dim = 2 * hidden_node_dim + edge_dim
+    self.linear_with_sigmoid = nn.Linear(z_dim, hidden_node_dim)
+    self.linear_with_softplus = nn.Linear(z_dim, hidden_node_dim)
+    self.batch_norm = nn.BatchNorm1d(hidden_node_dim) if batch_norm else None
+
+  def message_func(self, edges):
+    z = torch.cat(
+        [edges.src['x'], edges.dst['x'], edges.data['edge_attr']], dim=1)
+    gated_z = F.sigmoid(self.linear_with_sigmoid(z))
+    message_z = F.softplus(self.linear_with_softplus(z))
+    return {'gated_z': gated_z, 'message_z': message_z}
+
+  def reduce_func(self, nodes):
+    new_h = nodes.data['x'] + torch.sum(
+        nodes.mailbox['gated_z'] * nodes.mailbox['message_z'], dim=1)
+    return {'x': new_h}
+
+  def forward(self, dgl_graph):
+    """Update node representaions.
+
+    Parameters
+    ----------
+    dgl_graph: DGLGraph
+      DGLGraph for a batch of graphs. The graph expects that the node features
+      are stored in `ndata['x']`, and the edge features are stored in `edata['edge_attr']`.
+
+    Returns
+    -------
+    dgl_graph: DGLGraph
+      DGLGraph for a batch of updated graphs.
+    """
+    dgl_graph.update_all(self.message_func, self.reduce_func)
+    if self.batch_norm is not None:
+      dgl_graph.ndata['x'] = self.batch_norm(dgl_graph.ndata['x'])
+    return dgl_graph
+
+
+class CGCNN(nn.Module):
+  """Crystal Graph Convolutional Neural Network (CGCNN).
+
+  This model takes arbitary crystal structures as an input, and predict material properties
+  using the element information and connection of atoms in the crystal. If you want to get
+  some material properties which has a high computational cost like band gap in the case
+  of DFT, this model may be useful. This model is one of variants of Graph Convolutional
+  Networks. The main differences between other GCN models are how to construct graphs and
+  how to update node representations. This model defines the crystal graph from structures
+  using distances between atoms. The crystal graph is an undirected multigraph which is defined
+  by nodes representing atom properties and edges representing connections between atoms
+  in a crystal. And, this model updates the node representations using both neighbor node
+  and edge representations. Please confirm the detail algorithms from [1]_.
+
```

Comment:
Could you add a usage example to the docstring?

Reply:
I added!